### PR TITLE
Movie creation

### PR DESCRIPTION
**Pull request by j1nma**
## Summary
* I followed the Creado card to implement an end-point for the creation of movies, involving a MovieViewModel and UoW in MoviesRepository. Tag Helpers used in Movie View.

## Trello Card Link
https://trello.com/c/XqPtYDrN

## Screenshots
image

image

image

### CODE REVIEW

File: src/Controllers/MoviesController.cs [hunk 1]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
         }
         public IActionResult Index()
         {
-            return View ();
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+            var movies = _unitOfWork.MovieRepository.GetAll();
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 3]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+            return View(movies);
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: if we are using view models, you have to send MovieViewModel to the view instead Movie.

File: src/Controllers/MoviesController.cs [hunk 4]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+        }
+
+        public IActionResult Create()
+        {
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 5]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+            var model = new MovieViewModel();
+            return View(model);
+        }
+
+        [HttpPost]
+        public IActionResult Create(MovieViewModel model)
+        {
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 6]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+            if (ModelState.IsValid)
+            {
+                var movie = new Movie();
+                movie.ReleaseDate = model.ReleaseDate;
+                movie.Genre = model.Genre;
+                movie.Price = model.Price;
+                movie.Title = model.Title;
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs:46]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 7]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+
+                _unitOfWork.MovieRepository.Add(movie);
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 8]
```diff
@@ -19,7 +21,38 @@ public IUnitOfWork UnitOfWork
...
+                _unitOfWork.Complete();
+                return RedirectToAction("Index");
+            }
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

**Review by gzamudio — CHANGES_REQUESTED** [review]

File: src/Controllers/MoviesController.cs [hunk 9]
```diff
@@ -20,6 +23,39 @@ public IUnitOfWork UnitOfWork
 
         public IActionResult Index()
         {
+            var movies = _unitOfWork.MovieRepository.GetAll();
+
+            var moviesModels = new List<MovieViewModel>();
+
+            foreach (var m in movies)
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remember not to add empty lines inside methods.

File: src/Controllers/MoviesController.cs [hunk 10]
```diff
@@ -20,6 +23,39 @@ public IUnitOfWork UnitOfWork
...
+            {
+                var vm = new MovieViewModel { ID = m.ID, Title = m.Title, Genre = m.Genre, Price = m.Price, ReleaseDate = m.ReleaseDate };
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: this works fine, but to do things like this, we use LINQ. for next PR, google it.

**Review by gzamudio — APPROVED** [review]

File: src/Controllers/MoviesController.cs (hunk 10 above)
Comment (j1nma) [reply src/Controllers/MoviesController.cs]: carbon

**Pull request merged**